Added two other required columns and added indexs

ub0005 opened the pull request.

NISH1001 commented:
@ub0005 Looks good. But one thing:
Have we also updated the PATCH request for storage ids? (Just to make sure we have CRUD also...)

ub0005 commented:
> @ub0005 Looks good. But one thing: Have we also updated the PATCH request for storage ids? (Just to make sure we have CRUD also...)

Yes, @NISH1001 Patch requests take care of these additional fields automatically, no code modification is required, as we changed the models and schemas.

NISH1001 reviewed (commented):
Overall looks good....

xhagrg commented on src/app.py:
```diff
@@ -338,13 +338,19 @@ def upload_csv():
-        logger.error("uuid or name or sealed_state column values are empty!")
+    if data[CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS].isna().sum().sum() > 0:
+        logger.error(
+            "Any of the column "
+            + ",".join(CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS)
+            + " values are empty!"
```
Reuse variable.

ub0005 replied:
Reused Error Message

xhagrg commented on src/services/db/models.py:
```diff
@@ -32,18 +32,23 @@ class CatalogueItem(db.Model):
     transfer_destination = db.Column(db.String, nullable=True)
 
-    sealed_state = db.Column(db.String)
+    sealed_state = db.Column(db.String, index=True)
```
What's a sealed_state?

ub0005 replied:
The sealed state status indicates the file storage location. There are three sealed states. A sealed state indicates the file is in the cold storage/archived state. An unsealed state means that the file is in the hot storage state and readily available to be transferred. Unsealing state means that the file is being transitioned from cold storage to hot storage.

xhagrg commented:
LGTM.

xhagrg commented on src/constants.py:
```diff
@@ -35,3 +35,9 @@
 
 DATETIME_OLDEST = datetime.datetime(year=1970, month=1, day=1)
+
+ERROR_MSG_ANY_OF_THE_CATALOGUE_POST_MANDATORY_FIELDS_EMPTY = (
```
This can be just a simple variable in the app.py file rather than a config.py file entry.

ub0005 replied:
moved the variable to the app.py file

xhagrg commented on src/app.py:
```diff
@@ -338,13 +343,15 @@ def upload_csv():
-        logger.error("uuid or name or sealed_state column values are empty!")
+    if data[CONSTANTS.CATALOGUE_POST_MANDATORY_FIELDS].isna().sum().sum() > 0:
+        logger.error(
+            CONSTANTS.ERROR_MSG_ANY_OF_THE_CATALOGUE_POST_MANDATORY_FIELDS_EMPTY
```
It's not in the scope of constants now, right?

ub0005 replied:
yes Iksha, fixed and tested it.

xhagrg commented:
LGTM.
